Main/Source/char.cpp

```diff
@@ -2363,7 +2383,7 @@ void character::GetPlayerCommand()
   {
     game::DrawEverything();
 
-    if(game::GetDangerFound())
+    if(game::GetDangerFound() && !StateIsActivated(FEARLESS))
```

Comment:
So `fearless` is intended to be a double-edged sword? You will never panic, but you also aren't warned about dangerous enemies? I like it :)

Reply:
Micro-optimization, but I think checking for the fearless state before `game::GetDangerFound` might save some unnecessary cycles. Same in other places

Reply:
Yes, you become a bit too brave. :P

Do you mean only switching the order to '!StateIsActivated(FEARLESS) && game::GetDangerFound()', or moving the 'FEARLESS' check to 'GetDangerFound'?

Reply:
I meant switching the order. I don't think it's really important though. Optimization is just always on my mind.